lib/crawly/middlewares/unique_request.ex

```diff
@@ -1,9 +1,15 @@
 defmodule Crawly.Middlewares.UniqueRequest do
   @moduledoc """
-  Avoid scheduling requests for the same pages.
+  Avoid scheduling requests for the same pages. However if retry is requested
+  the request is still allowed.
   """
   require Logger
 
+  # Allow to re-schedule a request if retries are required
+  def run(%Crawly.Request{retries: retries} = request, state) when retries > 0 do
```

Comment:
The problem is that retries is introducing same requests several times. E.g. if we retry something it would mean that it has a stored fingerprint already. So we need to do something like a forced `store`. Otherwise, the retried requests can't really be unique.

Reply:
i see, that makes sense.Maybe the middleware should be renamed to convey its correct behaviour

Reply:
Yes, maybe. Any proposals? 

Also, I am not quite sure about this part:  `when retries > 0` e.g. it's not explicit ( E.g. it's not clear what is retry > 0 ). 

Should I add another field to a request, e.g. something like: is_retry_request: false/true, so it will be explicit?

Reply:
Coming up with a name is very hard because the two concerns are quite different. It's as if we're trying to fit a if-this-do-that into the name.

As suggested:
> A request key boolean flag (like `retry_if_fail`) that any middleware can set to true, and the worker simply checks if it is true or false.
Could something like this work? then the `UniqueRequest` middleware can just check if the request has this boolean flag and allow it to pass if so. A separate `Retry` middleware can then do the retry count increment and dropping